### PR TITLE
Refactor webhooks

### PR DESCRIPTION
**Pull request by ogenstad**

### CODE REVIEW

File: backend/infrahub/events/node_action.py [hunk 1]
```diff
@@ -33,13 +32,4 @@ def get_payload(self) -> dict[str, Any]:
         return self.data
 
     def get_messages(self) -> list[InfrahubMessage]:
-        return [
-            EventNodeMutated(
```
Comment (dgarros) [inline backend/infrahub/events/node_action.py]: I think we should keep this event for now.
we need to do some changes but completely removing it now means we'll need to redo everything shortly after

File: backend/infrahub/workflows/catalogue.py [hunk 2]
```diff
@@ -340,13 +340,42 @@
...
+)
+
+WEBHOOK_TRIGGER = WorkflowDefinition(
+    name="webhook-trigger-actions",
+    type=WorkflowType.USER,
+    module="infrahub.webhook.tasks",
+    function="trigger_webhooks",
+)
+
+WEBHOOK_TRIGGER = WorkflowDefinition(
```
Comment (dgarros) [inline backend/infrahub/workflows/catalogue.py]: WEBHOOK_TRIGGER is defined twice

File: backend/infrahub/webhook/tasks.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+        await service.workflow.submit_workflow(workflow=WEBHOOK_SEND, parameters={"model": model})
+
+
+@flow(name="webhook-setup-automations", flow_run_name="Configuration webhook automation and populate cache")
+async def configure_webhooks() -> None:
+    service = services.service
+    log = get_run_logger()
+
+    log.debug("Refreshing webhook configuration")
+    standard_webhooks = await service.client.all(kind=InfrahubKind.STANDARDWEBHOOK)
```
Comment (dgarros) [inline backend/infrahub/webhook/tasks.py]: We could use a protocol here

**Review by dgarros — APPROVED** [review]
Looks good to me

File: backend/infrahub/workflows/catalogue.py (hunk 2 above)
Comment (ogenstad) [reply backend/infrahub/workflows/catalogue.py]: Ah, yes thanks. I've removed the duplicate.

File: backend/infrahub/webhook/tasks.py (hunk 3 above)
Comment (ogenstad) [reply backend/infrahub/webhook/tasks.py]: Switched to protocols

File: backend/infrahub/events/node_action.py (hunk 1 above)
Comment (ogenstad) [reply backend/infrahub/events/node_action.py]: The main reason for opening this PR was actually to remove this as it generates a lot of extra noise and slows things down when a there's a high number of nodes that are mutated. I'm not sure I see the benefits of keeping it and what we'd need to redo later? (i.e. specifically related to this message)

Comment (ogenstad) [reply backend/infrahub/events/node_action.py]: I've added back the message but removed the flow from it, so at the moment it doesn't do anything aside from writing a debug log entry. But it's there for future use if we want it.

**Pull request merged**